File: min_distance.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
  int shortestDistance(vector<string>& wordsDict, string word1, string word2) {
    int w1Idx = 1e9;
    int w2Idx = -1e9;
    int distance = wordsDict.size();

    for (int i=0; i<wordsDict.size(); i++) {
      if (wordsDict[i] == word1) {
        w1Idx = i;
      } else if (wordsDict[i] == word2) {
        w2Idx = i;
      }

      if (w1Idx != 1e9 && w2Idx != -1e9) {
        std::cout << w1Idx << " : " << w2Idx << std::endl;
        distance = min(abs(w2Idx - w1Idx), distance);
      }
    }

    return distance;

  }
};

int main()
{
  Solution so;
  vector<string> words = {"practice", "makes", "perfect", "skill", "makes", "coding"};
  int result = so.shortestDistance(words, "perfect", "coding");
  std::cout << "Shortest distance : " << result << std::endl;
}

File: min_distance.js
/**
 * @param {string[]} wordsDict
 * @param {string} word1
 * @param {string} word2
 * @return {number}
 */
 var shortestDistance = function(wordsDict, word1, word2) {

  let w1Idx = null
  let w2Idx = null
  let minDistance = wordsDict.length
  for (let i=0; i<wordsDict.length; i++) {
      if (wordsDict[i] === word1) w1Idx = i
      if (wordsDict[i] === word2) w2Idx = i

      if (w1Idx !== null && w2Idx !== null) {
        minDistance = Math.min(minDistance, Math.abs(w1Idx - w2Idx))
      }
  }
  return minDistance
};

console.log(shortestDistance(["practice", "makes", "perfect", "coding", "makes"], "coding", "practice"))
console.log(shortestDistance(["practice", "makes", "perfect", "coding", "makes"], "makes", "coding"))